maven-archetypes/examples/src/test/resources/projects/basic/archetype.properties

```diff
@@ -16,4 +16,4 @@ package=it.pkg
 version=0.1
 groupId=archetype.it
 artifactId=basic
-targetPlatform=1.7
+targetPlatform=1.8
```

Comment:
Can we remove this line in the archetype.properties and instead just use the @java.version@?

Ditto on maven-archetypes/starter/src/main/resources/META-INF/maven/archetype-metadata.xml and maven-archetypes/starter/src/main/resources/archetype-resources/pom.xml

This should allow you to remove all references to targetPlatform everywhere.

Reply:
When I removed and used @java.version@, the target pom is picking up my local java version, not 1.8

Reply:
ready to merge

Reply:
I see